### PR TITLE
Address TypeError on XML with no tests - skip filter if there are no tests in the report

### PR DESCRIPTION
**Pull request by MjEatough**
Fixes error parsing results files if an XML has no tests. When running multiple tests the Cypress framework can output an empty XML alongside the XMLs for each suite.  Error is: 

```
TypeError: Cannot read properties of undefined (reading 'filter')
    at getTestResult (/home/runner/work/cypress-tests/cypress-tests/node_modules/test-results-parser/src/parsers/junit.js:80:36)
    at Object.parse (/home/runner/work/cypress-tests/cypress-tests/node_modules/test-results-parser/src/parsers/junit.js:91:10)
...
```

When parsing an XML that looks like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Cypress Tests 2022-10-13T19:34:52.973Z" time="0.0000" tests="0" failures="0">
</testsuites>
```

Fix skips filtering on the XML if there are no tests present.

### CODE REVIEW

File: src/parsers/junit.js [hunk 1]
```diff
@@ -76,10 +76,12 @@ function getTestResult(json) {
   result.total = result.total - result.skipped;
   result.passed = result.total - result.failed - result.errors;
   result.duration = rawResult["@_time"] * 1000;
-  const rawSuites = rawResult["testsuite"];
-  const filteredSuites = rawSuites.filter(suite => suite.testcase);
-  for (let i = 0; i < filteredSuites.length; i++) {
-    result.suites.push(getTestSuite(filteredSuites[i]));
+  if (result.total > 0 && result.skpped == 0) { // Don't filter if there are no tests
```
Comment (ASaiAnudeep) [inline src/parsers/junit.js:79]: - `result.skpped` is a typo
- there are chances that a test suite can contain skipped tests along with the others.

Comment (MjEatough) [reply src/parsers/junit.js]: Fixed compare and verified unit tests pass with changes.

**Comment by ASaiAnudeep** [comment]
Could you please add a new unit test to verify your changes.

File: src/parsers/junit.js [hunk 2]
```diff
@@ -91,9 +91,11 @@ function getTestResult(json) {
   result.passed = result.total - result.failed - result.errors;
   result.duration = rawResult["@_time"] * 1000;
   const rawSuites = rawResult["testsuite"];
-  const filteredSuites = rawSuites.filter(suite => suite.testcase);
-  for (let i = 0; i < filteredSuites.length; i++) {
-    result.suites.push(getTestSuite(filteredSuites[i]));
+    if (!(typeof rawSuites === "undefined")) { // Don't filter if there are no testsuite objects
+    const filteredSuites = rawSuites.filter(suite => suite.testcase);
```
Comment (ASaiAnudeep) [inline src/parsers/junit.js:95]:
```suggestion
      const filteredSuites = rawSuites.filter(suite => suite.testcase);
```

File: src/parsers/junit.js [hunk 3]
```diff
@@ -91,9 +91,11 @@ function getTestResult(json) {
...
+    for (let i = 0; i < filteredSuites.length; i++) {
```
Comment (ASaiAnudeep) [inline src/parsers/junit.js:96]:
```suggestion
      for (let i = 0; i < filteredSuites.length; i++) {
```

File: src/parsers/junit.js [hunk 4]
```diff
@@ -91,9 +91,11 @@ function getTestResult(json) {
...
+      result.suites.push(getTestSuite(filteredSuites[i]));
```
Comment (ASaiAnudeep) [inline src/parsers/junit.js:97]:
```suggestion
        result.suites.push(getTestSuite(filteredSuites[i]));
```

File: src/parsers/junit.js [hunk 5]
```diff
@@ -91,9 +91,11 @@ function getTestResult(json) {
   result.passed = result.total - result.failed - result.errors;
   result.duration = rawResult["@_time"] * 1000;
   const rawSuites = rawResult["testsuite"];
-  const filteredSuites = rawSuites.filter(suite => suite.testcase);
-  for (let i = 0; i < filteredSuites.length; i++) {
-    result.suites.push(getTestSuite(filteredSuites[i]));
+    if (!(typeof rawSuites === "undefined")) { // Don't filter if there are no testsuite objects
```
Comment (ASaiAnudeep) [inline src/parsers/junit.js:94]:
```suggestion
  if (!(typeof rawSuites === "undefined")) { // Don't filter if there are no testsuite objects
```

**Pull request merged**